### PR TITLE
feat: Added implementation of ISS-CAM

### PR DESCRIPTION
**Pull request by frgfm**
This PR introduces a variant of SS-CAM using integration rather than summation by:
- implementing the `ISSCAM`class
- updating documentation
- updating unittests
- updating the sample script

### CODE REVIEW

File: torchcam/cams/cam.py [hunk 1]
```diff
@@ -353,3 +353,100 @@ def _get_weights(self, class_idx, scores=None):
...
+        upsampled_a = upsampled_a.squeeze(0).unsqueeze(1)
+
+        # Initialize weights
+        weights = torch.zeros(upsampled_a.shape[0], dtype=upsampled_a.dtype).to(device=upsampled_a.device)
+
+        # Disable hook updates
+        self._hooks_enabled = False
+
+        for _idx in range(self.num_samples):
+            noisy_m = self._input * (upsampled_a +
```
Comment (r0cketr1kky) [inline torchcam/cams/cam.py:427]: I guess this should be different
We would need to initialize `noisy_m` to 0 and a variable `c` to `1 / self.num_samples `
We want to calculate `noisy_m += self._input * upsampled_a * c * _idx`

Comment (frgfm) [reply torchcam/cams/cam.py]: Good catch, that should be solved now!

File: torchcam/cams/cam.py [hunk 2]
```diff
@@ -422,18 +427,18 @@ def _get_weights(self, class_idx, scores=None):
 
         # Disable hook updates
         self._hooks_enabled = False
+        fmap = 0
 
         for _idx in range(self.num_samples):
-            noisy_m = (_idx + 1) / self.num_samples * self._input * (upsampled_a +
-                                     self._distrib.sample(self._input.size()).to(device=self._input.device))
+            fmap += (_idx + 1) / self.num_samples * self._input * upsampled_a
```
Comment (r0cketr1kky) [inline torchcam/cams/cam.py:433]: This should probably be `fmap += ((_idx + 1) / self.num_samples) * self._input * upsampled_a`
Not sure if this is the execution that's carried out :-)

File: torchcam/cams/cam.py [hunk 3]
```diff
@@ -422,18 +427,18 @@ def _get_weights(self, class_idx, scores=None):
 
         # Disable hook updates
         self._hooks_enabled = False
+        fmap = 0
```
Comment (r0cketr1kky) [inline torchcam/cams/cam.py:431]: Add `start_count = 0.1`

**Pull request merged**